sdk/eventhub/azure-eventhubs/azure/eventhub/_producer_client.py

```diff
@@ -176,21 +172,25 @@ def send(self, event_data, **kwargs):
                 :caption: Sends event data
 
         """
-        partition_id = kwargs.pop("partition_id", None) or ALL_PARTITIONS
+        partition_id = event_data_batch._partition_id or ALL_PARTITIONS  # pylint:disable=protected-access
         send_timeout = kwargs.pop("timeout", None)
         try:
-            self._producers[partition_id].send(event_data, **kwargs)
+            self._producers[partition_id].send(event_data_batch, timeout=send_timeout)
         except (KeyError, AttributeError, EventHubError):
             self._start_producer(partition_id, send_timeout)
-            self._producers[partition_id].send(event_data, **kwargs)
+            self._producers[partition_id].send(event_data_batch, timeout=send_timeout)
 
-    def create_batch(self, max_size=None):
-        # type:(int) -> EventDataBatch
+    def create_batch(self, **kwargs):
+        # type:(Any) -> EventDataBatch
         """
         Create an EventDataBatch object with max size being max_size.
         The max_size should be no greater than the max allowed message size defined by the service side.
 
-        :param int max_size: The maximum size of bytes data that an EventDataBatch object can hold.
+        :keyword str partition_id: The specific partition ID to send to. Default is None, in which case the service
+         will assign to all partitions using round-robin.
+        :keyword str partition_key: With the given partition_key, event data will land to
```

Comment:
Full disclosure:  I could absolutely google for docs and figure out a difference between these two vars.

But at a glance?  I have no idea what partition_key does, more more precisely, how it is different from partition_id.  The current description seems to imply it's service-side instead of client-side round robin?  But still not sure how a key relates to an ID, even with looking at the portal alongside this.

Reply:
It's addressed in the official documentation:
https://docs.microsoft.com/en-ca/azure/event-hubs/event-hubs-features#partitions